lib/palletjack/ipa.rb

```diff
@@ -0,0 +1,273 @@
+require 'curb'
+require 'resolv'
+require 'json'
+
+class PalletJack
+  class Ipa
+
+    # Helper class for the <tt>PalletJack::Ipa::Command</tt> class
+    #
+    # This class is initialized with a payload dict that resembles
+    # the data to be sent to the JSON-RPC endpoint. The class will
+    # take care of session management troughout the lifetime of the
+    # application.
+    #
+    # Authentication is done trough GSSAPI. It is assumed that whoever
+    # running the script has a valid kerberos credential.
+    #
+    # The endpoint is automatically determined and it is assumed that
+    # the FQDN of the current machine is the same as the IPA domain.
+    #
+    # See the Red Hat documentation about JSON-RPC for more info:
+    # https://access.redhat.com/articles/2728021
+    #
+
+    class Request
+
+      # Save the curl object, when we login we will recive a session
+      # cookie that is saved within the @@curl objects cookie cache.
+      @@curl = nil
+      @@session_init = false
+
+      @@ipa_url = nil
+
+      # Perform a HTTP request against the JSON-RPC endpoint of a IPA server.
+      def initialize(payload, debug = false)
```

Comment:
The debug feature should be activated by setting a class attribute instead of having it as a visible part of the object creation API.

```
def self.debug=(maybe)
  @@debug = maybe
end

def self.debug
  @@debug ||= false
end

private

def debug?
  self.class.debug
end
```